container.go

```diff
@@ -495,7 +495,7 @@ func (c *Container) Execute(args ...string) ([]byte, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if err := c.makeSure(isNotDefined); err != nil {
+	if err := c.makeSure(isDefined); err != nil {
```

Comment:
So I think the "isNotDefined" was actually intentional here. I ran into this earlier when fixing a bunch of issues with the go-lxc tests. Execute, per its description, was meant to run a command inside a temporary, throw-away container, not using a defined LXC container.

@caglar10ur may have more background on why that was done this way.

Reply:
I agree that it'd have been better if Execute() behaved like lxc-execute, unfortunately that's not how it's been implemented and I'm worried that reversing that logic now will break users.

Reply:
I think - as @stgraber said this was meant to run a command inside a temporary, throw-away container, not using a defined LXC container.

How about adding ExecuteInContainer for executing something inside an already defined container?

Reply:
That would be the same as StartExecute or whatever it's called that @brauner added though right?

Reply:
Ah yep, just saw that you merged it :)

Reply:
Though maybe we can rename it to ExecuteInContainer now that it is new and there are no users of that function, if it makes more sense to you.

Reply:
Then perhaps we should just delete this then: the code simply will not work as-is without a defined container, because it calls lxc-execute, which itself requires a config file, which will not be present.

Reply:
That's why I added the code right below this check which on 2.1+ will create and then remove a temporary config, I believe this did fix the tests on recent liblxc at least.

Reply:
Maybe we should error out on <= 2.1 then? Because without that, it won't work.

Reply:
I'm confused, I only started seeing this failing with 2.1+ so thought it was a behavior change and that 2.0 and earlier didn't need this particular change.

If 2.0 also needs this now, then we should just change the version check to cover it too.

Reply:
Ok, so we should change go-lxc to just always create the temp config. It's a hack but to be fair, this entire function is a hack so whatever :)